pmd-java/src/main/java/net/sourceforge/pmd/lang/java/metrics/impl/internal/CognitiveComplexityVisitor.java

```diff
@@ -264,7 +264,8 @@ public Object visit(ASTPrimaryPrefix node, Object data) {
             if (child instanceof ASTName) {
                 ASTName name = (ASTName) child;
                 if (name.getNameDeclaration() instanceof MethodNameDeclaration) {
-                    state.methodCall(name.getNameDeclaration().getName());
+                    ASTMethodDeclaration parent = (ASTMethodDeclaration) name.getNameDeclaration().getNode().getParent();
```

Comment:
... then compare this `parent`  node against the saved one

Reply:
Done! Btw I do identity comparison (`==`)  because I am not sure if `equals`/`hashCode` are defined for nodes. Let me know if it works.

Reply:
Yes this is the way :)